native/common/jp_exception.cpp

```diff
@@ -432,6 +425,25 @@ void JPypeException::toJava(JPContext *context)
 			frame.ThrowNew(context->m_NoSuchMethodError.get(), mesg.c_str());
 			return;
 		}
+
+		if (m_Type == JPError::_python_error)
+		{
+			JPPyCallAcquire callback;
+			JP_TRACE("Python exception");
+			convertPythonToJava(context);
+			return;
+		}
+
+		if (m_Type == JPError::_python_exc)
+		{
+			JPPyCallAcquire callback;
+			// All others are Python errors
+			JP_TRACE(Py_TYPE(m_Error.l)->tp_name);
+			PyErr_SetString((PyObject*) m_Error.l, mesg.c_str());
+			convertPythonToJava(context);
+			return;
+		}
+
```

Comment:
This section is to correct a bug in which Python type information was lost when returning to Python for one specific path.